cli/pcluster/configure/easyconfig.py

```diff
@@ -199,56 +198,80 @@ def _reset_config_params(section, parameters_to_remove):
             vpc_parameters.update(
-                automate_vpc_with_subnet_creation(_choose_network_configuration(scheduler), min_subnet_size)
+                automate_vpc_with_subnet_creation(
+                    _choose_network_configuration(scheduler, qualified_azs), min_subnet_size
+                )
             )
         else:
             default_vpc = vpc_section.get_param_value("vpc_id")
             vpc_id = prompt_iterable(
                 "VPC ID",
                 vpc_list,
-                default_value=default_vpc if default_vpc in [vpc_entry[0] for vpc_entry in vpc_list] else None,
+                default_value=default_vpc if default_vpc in [vpc_entry["id"] for vpc_entry in vpc_list] else None,
             )
             vpc_parameters["vpc_id"] = vpc_id
             subnet_list = vpc_and_subnets["vpc_subnets"][vpc_id]
-            if not subnet_list or (
-                prompt("Automate Subnet creation? (y/n)", lambda x: x in ("y", "n"), default_value="y") == "y"
+            qualified_master_subnets = _filter_subnets_by_azs(subnet_list, qualified_azs["azs_for_master_type"])
+            if "azs_for_compute_type" in qualified_azs.keys():
+                qualified_compute_subnets = _filter_subnets_by_azs(subnet_list, qualified_azs["azs_for_compute_type"])
+            else:
+                # Special case of awsbatch, where compute instance type is not specified
+                qualified_compute_subnets = subnet_list
+            if (
+                not qualified_master_subnets
+                or not qualified_compute_subnets
+                or (prompt("Automate Subnet creation? (y/n)", lambda x: x in ("y", "n"), default_value="y") == "y")
             ):
+                # Start auto subnets creation in the absence of qualified subnets.
+                # Otherwise, user selects between manual and automate subnets creation
                 vpc_parameters.update(
-                    automate_subnet_creation(vpc_id, _choose_network_configuration(scheduler), min_subnet_size)
+                    automate_subnet_creation(
+                        vpc_id, _choose_network_configuration(scheduler, qualified_azs), min_subnet_size
+                    )
                 )
             else:
-                vpc_parameters.update(_ask_for_subnets(subnet_list, vpc_section))
+                vpc_parameters.update(
+                    _ask_for_subnets(subnet_list, vpc_section, qualified_master_subnets, qualified_compute_subnets)
+                )
     return vpc_parameters
 
 
-def _ask_for_subnets(subnet_list, vpc_section):
-    available_subnets = [subnet_entry[0] for subnet_entry in subnet_list]
+def _filter_subnets_by_azs(subnet_list, qualified_azs):
+    return [subnet_entry for subnet_entry in subnet_list if subnet_entry["availability_zone"] in qualified_azs]
+
+
+def _ask_for_subnets(subnet_list, vpc_section, qualified_master_subnets, qualified_compute_subnets):
```

Comment:
code to prompt for master vs compute subnets looks almost identical. Let's see how we can abstract this functionality into a single function to reduce code duplication.